torch/_inductor/lowering.py

```diff
@@ -225,7 +225,6 @@ def _register_foreach_lowering(aten_fn, decomp_fn):
 
     @functools.wraps(decomp_fn)
     def wrapped(*args, **kwargs):
-        assert len(args) <= 2
```

Comment:
@mlazos I could change this to assert len(args) <= 3 to account for the alpha, but when I look into the lowered code, the alpha is nowhere to be found, so there might be a bigger problem here that should be addressed.

In this current PR, the alpha doesn't really matter, but I feel like either this assert should be turned to 3 OR something bigger may be happening. In the other make_pointwise code, alpha is handled by multiplying with the last set of inputs. I'm curious why we don't do that in the grouping code in make_pointwise_foreach.

Reply:
This only supports a kwarg alpha it looks like, I don't think we can arbitrarily set the len(args) <= 3, there needs to be additional handling. I think the only alpha that is supported is through kwargs, I didn't even know there was a non-kwarg one tbh.

Reply:
I re-added the assert, named the alpha, and tried running test_adagrad. It still fails on the assert with alpha=1 as the 3rd argument.
```
  File "/data/users/janeyx/pytorch/torch/_inductor/lowering.py", line 228, in wrapped
    assert len(args) <= 2
torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
LoweringException: AssertionError: 
  target: aten._foreach_add.Tensor
  args[0]: [TensorBox(StorageBox(
    InputBuffer(name='arg8_1', layout=FixedLayout('cpu', torch.float32, size=[], stride=[]))
  )), TensorBox(StorageBox(
    InputBuffer(name='arg9_1', layout=FixedLayout('cpu', torch.float32, size=[], stride=[]))
  )), TensorBox(StorageBox(
    InputBuffer(name='arg10_1', layout=FixedLayout('cpu', torch.float32, size=[], stride=[]))
  )), TensorBox(StorageBox(
    InputBuffer(name='arg11_1', layout=FixedLayout('cpu', torch.float32, size=[], stride=[]))
  ))]
  args[1]: TensorBox(StorageBox(
    Pointwise(
      'cpu',
      torch.float32,
      def inner_fn(index):
          tmp0 = ops.constant(1.0, torch.float32)
          return tmp0
      ,
      ranges=[],
      origin_node=full_default,
      origins={full_default}
    )
  ))
  args[2]: 1.0
```

Reply:
This code works for me: https://gist.github.com/mlazos/77b1cb4cf69b147afebacbdef4469855

Reply:
Resolved--I missed a * in the native_functions overload which made the alpha no longer kwarg-only. Thanks for pointing that out!